### DISCUSSION TITLE
Detailed Material Documentation

### DISCUSSION THREAD

**Discussion by DasLixou** [material-components/material-web#4434] [Q&A] [answered]
Hello!
I know this is not really the ideal place to ask this question but this is the best place I could find. I am trying to implement some of Material 3 Components in C# but struggle a lot at following the components specs on m3.material.io - The base layout description and coloring with tokens is good, don't get me wrong. But there are a lot of questions I have. For example: When should i use a shadow, how strong should it be? How should the animations be like?  For example the circle like animation on FAButtons - i cant find any information about those and how to implement them. Also: Where should the click / interaction region for a button be - should the text in a NavigationRail also navigate to the page? I searched all those things up but really can't find anything properly documented about them. I also tried to extract some knowledge from this project and the figma implementation but this is also really hard.
So I just wanted to ask if there is any proper resource about those questions which would help me with implementing Material 3.
Thanks :)

**Comment by DasLixou** [comment]
Ah yes, and also: There is nowhere documented which Weight etc. should be used for Icons in NavigationRailDestinations or FABs

**Comment by asyncliz** [comment]
Unfortunately this is a known gap in material.io's documentation. Other than looking at implementing libraries like MWC and MDC Android and asking questions, there isn't a good resource for many of these undocumented aspects of Material.

**Reply by DasLixou** [reply]
So should I just like guess what they‘re supposed to look like and function?

**Reply by DasLixou** [reply]
Do you then here also just try to match the behavior of e.g. flutters impl?

**Reply by asyncliz** [reply, marked as answer]
There is more internal documentation and design reviews that we reference. Unfortunately they're not published externally

**Reply by DasLixou** [reply]
Oh that's sad. Sounds a bit like they don't even want it to be implemented from 3rd developers.. guess I'll have to dig trough the source code then..

**Reply by DasLixou** [reply]
So i was digging through the documentation here in this repo and found some interesting docs about things like elevation, ripple and focus ring. Will they also be documented on m3.material.io in the future? And it says `API Documentation (coming soon)` everywhere. Is this what I am looking for?

**Reply by DasLixou** [reply]
And when you say "..other than answering questions..", may it then be possible to get my questions mentioned above answered? Thanks
